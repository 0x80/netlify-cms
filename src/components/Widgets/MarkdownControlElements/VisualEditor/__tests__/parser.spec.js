import { Schema } from "prosemirror-model";
import { schema } from "prosemirror-markdown";

const makeParser = require("../parser");

const testSchema = new Schema({
  nodes: schema.nodeSpec,
  marks: schema.markSpec,
});
const parser = makeParser(testSchema);

describe("Compile markdown to Prosemirror document structure", () => {
  it("should compile simple markdown", () => {
    const value = `
# H1

sweet body
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile a markdown ordered list", () => {
    const value = `
# H1

1. yo
2. bro
3. fro
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile bulleted lists", () => {
    const value = `
# H1

* yo
* bro
* fro
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile multiple header levels", () => {
    const value = `
# H1

## H2

### H3
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile horizontal rules", () => {
    const value = `
# H1

---

blue moon
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile horizontal rules", () => {
    const value = `
# H1

---

blue moon
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile hard breaks (double space)", () => {
    const value = `
blue moon  
footballs
`;
    expect(parser(value)).toMatchSnapshot();
  });

  it("should compile images", () => {
    const value = `
![super](duper.jpg)
`;
    expect(parser(value)).toMatchSnapshot();
  });
});
