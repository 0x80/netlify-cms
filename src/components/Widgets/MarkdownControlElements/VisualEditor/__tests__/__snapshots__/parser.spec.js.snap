// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compile markdown to Prosemirror document structure should compile a markdown ordered list 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "order": 1,
        "tight": true,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "yo",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "list_item",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "bro",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "list_item",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "fro",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "list_item",
        },
      ],
      "type": "ordered_list",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile bulleted lists 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "tight": false,
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "yo",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "list_item",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "bro",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "list_item",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "fro",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "list_item",
        },
      ],
      "type": "bullet_list",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile hard breaks (double space) 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "blue moon",
          "type": "text",
        },
        Object {
          "type": "hard_break",
        },
        Object {
          "text": "footballs",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile horizontal rules 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "type": "horizontal_rule",
    },
    Object {
      "content": Array [
        Object {
          "text": "blue moon",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile horizontal rules 2`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "type": "horizontal_rule",
    },
    Object {
      "content": Array [
        Object {
          "text": "blue moon",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile images 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "alt": "super",
            "src": "duper.jpg",
            "title": null,
          },
          "type": "image",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile multiple header levels 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 2,
      },
      "content": Array [
        Object {
          "text": "H2",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "attrs": Object {
        "level": 3,
      },
      "content": Array [
        Object {
          "text": "H3",
          "type": "text",
        },
      ],
      "type": "heading",
    },
  ],
  "type": "doc",
}
`;

exports[`Compile markdown to Prosemirror document structure should compile simple markdown 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "content": Array [
        Object {
          "text": "H1",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "content": Array [
        Object {
          "text": "sweet body",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
